parsec/core/fs/workspacefs/workspacefs.py

```diff
@@ -186,12 +186,55 @@ def workspace_name(self) -> str:
     async def move(self, source: AnyPath, destination: AnyPath):
         source = FsPath(source)
         destination = FsPath(destination)
+        real_destination = destination
         if source.parent == destination.parent:
             return await self.rename(source, destination)
-        # TODO - reference implementation:
-        # https://github.com/python/cpython/blob/3.7/Lib/shutil.py#L525
+        try:
+            if await self.is_dir(destination):
+                real_destination = destination.joinpath(source.name)
+                if await self.path_info(real_destination):
```

Comment:
This make me realize I forgot to add the [exists](https://docs.python.org/3.5/library/pathlib.html#pathlib.Path.exists) method, feel free to add it :)

Reply:
Done!